internal/transports/p2p/peer/checkpoint_test.go

```diff
@@ -0,0 +1,280 @@
+package peer
+
+import (
+	"testing"
+
+	"github.com/bitcoin-sv/block-headers-service/domains"
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/internal/tests/fixtures"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckpointCreationLastReached(t *testing.T) {
+	// setup
+	checkpoints := chaincfg.MainNetParams.Checkpoints
+	testLogger := zerolog.Nop()
+
+	cases := map[string]struct {
+		checkpoints         []chaincfg.Checkpoint
+		tipHeight           int32
+		expectedLastReached bool
+	}{
+		"last checkpoint not reached at height 0": {
+			checkpoints:         checkpoints,
+			tipHeight:           0,
+			expectedLastReached: false,
+		},
+		"last checkpoint not reached at height 810 000": {
+			checkpoints:         checkpoints,
+			tipHeight:           810000,
+			expectedLastReached: false,
+		},
+		"last checkpoint reached at height 999 999 999": {
+			checkpoints:         checkpoints,
+			tipHeight:           999_999_999,
+			expectedLastReached: true,
+		},
+		"last checkpoint reached if checkpoints list is empty": {
+			checkpoints:         make([]chaincfg.Checkpoint, 0),
+			tipHeight:           0,
+			expectedLastReached: true,
+		},
+	}
+
+	for name, params := range cases {
+		t.Run(name, func(t *testing.T) {
+			// given:
+			chckPoint := newCheckpoint(params.checkpoints, params.tipHeight, &testLogger)
+
+			// expect:
+			assert.Equal(t, params.expectedLastReached, chckPoint.LastReached())
+		})
+	}
+
+}
+
+func TestCheckpointUseTip(t *testing.T) {
```

Comment:
If we remove `UseTip` method, we can just test `next` method here with the same test cases.

Reply:
I don't want to test method next, because it is intentionally private for this component, and it shouldn't bother us how it is working, so it's simpler to make any refactoring or splitting or anything.

Reply:
I think the method `findNextCheckpoint` is a method that should be private and should not bother us, but the `next()` method could be public. But it's just my opinion. Maybe @chris-4chain has an opinion about this as well? 😄

Reply:
method removed